python/cudf/cudf/core/column/column.py

```diff
@@ -3253,6 +3256,19 @@ def as_column(
         return as_column(arbitrary, nan_as_null=nan_as_null, dtype=dtype)
 
 
+def _column_from_cuda_array_interface(
+    arbitrary, nan_as_null, dtype, data_ptr_exposed=no_default
+):
+    column = ColumnBase.from_cuda_array_interface(
+        arbitrary, data_ptr_exposed=data_ptr_exposed
+    )
+    if nan_as_null is not False:
+        column = column.nans_to_nulls()
+    if dtype is not None:
+        column = column.astype(dtype)
```

Comment:
Could you instead keep these checks in the `elif hasattr(arbitrary, "__cuda_array_interface__")` and just call `ColumnBase.from_cuda_array_interface(...)` where needed instead? I'd like to remove these `if nan_as_null is not False` and `if dtype is not None` from `as_column` in the future and having these checks only in 1 function would help toward that.